core/coins/format.go

```diff
@@ -3,104 +3,16 @@ package coins
 import (
 	"fmt"
 	"regexp"
-	"sort"
 	"strings"
 
-	bankv1beta1 "cosmossdk.io/api/cosmos/bank/v1beta1"
 	basev1beta1 "cosmossdk.io/api/cosmos/base/v1beta1"
-	"cosmossdk.io/math"
 )
 
-const emptyCoins = "zero"
-
 // Amount can be a whole number or a decimal number. Denominations can be 3 ~ 128
 // characters long and support letters, followed by either a letter, a number or
 // a separator ('/', ':', '.', '_' or '-').
 var coinRegex = regexp.MustCompile(`^(\d+(\.\d+)?)([a-zA-Z][a-zA-Z0-9\/\:\._\-]{2,127})$`)
 
-// formatCoin formats a sdk.Coin into a value-rendered string, using the
-// given metadata about the denom. It returns the formatted coin string, the
-// display denom, and an optional error.
-func formatCoin(coin *basev1beta1.Coin, metadata *bankv1beta1.Metadata) (string, error) {
-	coinDenom := coin.Denom
-
-	// Return early if no display denom or display denom is the current coin denom.
-	if metadata == nil || metadata.Display == "" || coinDenom == metadata.Display {
-		vr, err := math.FormatDec(coin.Amount)
-		return vr + " " + coin.Denom, err
-	}
-
-	dispDenom := metadata.Display
-
-	// Find exponents of both denoms.
-	var coinExp, dispExp uint32
-	foundCoinExp, foundDispExp := false, false
-	for _, unit := range metadata.DenomUnits {
-		if coinDenom == unit.Denom {
-			coinExp = unit.Exponent
-			foundCoinExp = true
-		}
-		if dispDenom == unit.Denom {
-			dispExp = unit.Exponent
-			foundDispExp = true
-		}
-	}
-
-	// If we didn't find either exponent, then we return early.
-	if !foundCoinExp || !foundDispExp {
-		vr, err := math.FormatInt(coin.Amount)
-		return vr + " " + coin.Denom, err
-	}
-
-	dispAmount, err := math.LegacyNewDecFromStr(coin.Amount)
-	if err != nil {
-		return "", err
-	}
-
-	if coinExp > dispExp {
-		dispAmount = dispAmount.Mul(math.LegacyNewDec(10).Power(uint64(coinExp - dispExp)))
-	} else {
-		dispAmount = dispAmount.Quo(math.LegacyNewDec(10).Power(uint64(dispExp - coinExp)))
-	}
-
-	vr, err := math.FormatDec(dispAmount.String())
-	return vr + " " + dispDenom, err
-}
-
-// FormatCoins formats Coins into a value-rendered string, which uses
-// `formatCoin` separated by ", " (a comma and a space), and sorted
-// alphabetically by value-rendered denoms. It expects an array of metadata
-// (optionally nil), where each metadata at index `i` MUST match the coin denom
-// at the same index.
-func FormatCoins(coins []*basev1beta1.Coin, metadata []*bankv1beta1.Metadata) (string, error) {
```

Comment:
Can we get a changelog in core, as this is breaking?

Reply:
pushed, thank you!!